proc-macros/src/lib.rs

```diff
@@ -111,212 +110,278 @@ pub fn rpc_client_api(input_token_stream: TokenStream) -> TokenStream {
+/// }
+///
+/// impl<T> RpcClient for T where T: SubscriptionClient {}
+/// ```
+///
+/// ## Attributes
+///
+/// ### `rpc` attribute
+///
+/// `rpc` attribute is applied to a trait in order to turn it into an RPC implementation.
+///
+/// **Arguments:**
+///
+/// - `server`: generate `<Trait>Server` trait for the server implementation.
+/// - `client`: generate `<Trait>Client` extension trait that makes RPC clients to invoke a concrete RPC
+///   implementation methods conveniently.
+/// - `namespace`: add a prefix to all the methods and subscriptions in this RPC. For example, with namespace
+///   `foo` and method `spam`, the resulting method name will be `foo_spam`.
+///
+/// **Trait requirements:**
+///
+/// Trait wrapped with an `rfc` attribute **must not**:
+///
+/// - have associated types or constants;
+/// - have Rust methods not marked with either `method` or `subscription` attribute;
+/// - be empty.
+///
+/// At least one of the `server` or `client` flags must be provided, otherwise the compilation will err.
+///
+/// ### `method` attribute
+///
+/// `method` attribute is used to define an RPC method.
+///
+/// **Arguments:**
+///
+/// - `name` (mandatory): name of the RPC method. Does not have to be the same as the Rust method name.
+///
+/// **Method requirements:**
+///
+/// Rust method marked with `method` attribute, **may**:
+///
+/// - be either `async` or not;
+/// - have input parameters or not;
+/// - have return value or not (in the latter case, it will be considered a notification method).
+///
+/// ### `subscription` attribute
+///
+/// **Arguments:**
+///
+/// - `name` (mandatory): name of the RPC method. Does not have to be the same as the Rust method name.
+/// - `unsub` (mandatory): name of the RPC method to unsubscribe from the subscription. Must not be the same as `name`.
+/// - `item` (mandatory): type of items yielded by the subscription. Note that it must be the type, not string.
+///
+/// **Method requirements:**
+///
+/// Rust method marked with `subscription` attribute **must**:
+///
+/// - be synchronous;
+/// - not have return value.
+///
+/// Rust method marked with `subscription` attribute **may**:
+///
+/// - have input parameters or not.
+///
+/// ## Full workflow example
+///
+/// ```rust
+/// //! Example of using proc macro to generate working client and server.
+///
+/// use std::net::SocketAddr;
+///
+/// use futures_channel::oneshot;
+/// use jsonrpsee::{ws_client::*, ws_server::WsServerBuilder};
+///
+/// // RPC is moved into a separate module to clearly show names of generated entities.
+/// mod rpc_impl {
+///     use jsonrpsee::{proc_macros::rpc, types::async_trait, ws_server::SubscriptionSink};
+///
+///     // Generate both server and client implementations, prepend all the methods with `foo_` prefix.
+///     #[rpc(client, server, namespace = "foo")]
+///     pub trait Rpc {
+///         #[method(name = "foo")]
+///         async fn async_method(&self, param_a: u8, param_b: String) -> u16;
```

Comment:
I haven't read the code but is it possible to name this method to whatever I want such as `async fn get_me_some_food(&self,  portions: u8, dish: String)` in the case instead of `async_method?